### PR TITLE
Prevent memberships to use cache sites data

### PR DESCRIPTION
**Pull request by millerf**
Fixes https://github.com/Automattic/gold/issues/174

See p81Rsd-1u6-p2

## Proposed changes:

* When getting data from cache sites, use proper source of truth 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Self-hosted
* Create a JN site or an atomic site
* Turn OFF Jetpack Sync
* Make sure Paid-Newsletters, tiers or any membership blocks to work as expected

#### WPCOM
* Make sure paid-newsletters/tiers keep working as expected

### CODE REVIEW

**Review by chrismccluskey — COMMENTED** [review]
The code LGTM, although I did find one issue: paying for an upgrade on a newsletter tier by entering my email address again after purchasing a lower tier failed. I had disabled sync via the jetpack debugger, and everything worked great, aside from what I mentioned. After purchasing an upgraded tier failed, I enabled sync, scheduled a full sync and opened a new private window. In the new private window I recreated the test with a new user and the payment succeeded.

To be clear, when testing I tried my saved card ending in 4242 first, which failed. After that failure I tried adding a new card ending in 4242 which also failed. Even though the form itself said (test card) and I was in test mode. Only enabling sync and syncing made this work properly.

**Comment by millerf** [comment]
@chrismccluskey I updated the missing parts. There was quite a lot missing...

**Comment by chrismccluskey** [comment]
I'm still encountering an issue when upgrading a user. For whatever reason, the test card declines when sync isn't enabled.

**Review by n3f — DISMISSED** [review]
This all worked for me with a small exception (for an unrelated issue).

The issue was that the upgrade was failing when the pro-rata amount resolved to something less than the stripe minimum amount ($0.50).

File: projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php
Comment (simison) [inline projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:132]: In previous code you had `$has_access = true;` for this case?

Comment (simison) [reply projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-token-subscription-service.php:132]: Ah sorry I missed it combines with default below, ignore.

**Review by simison — APPROVED** [review]
Re-approving after team's [own](https://github.com/Automattic/jetpack/pull/33502#pullrequestreview-1683361254) review.

I only reviewed that the latest changes after team's approval are solid.

https://github.com/Automattic/jetpack/pull/33502/commits/cf8c83d408c1b9bdad1d3e540c0f168c302e5514
https://github.com/Automattic/jetpack/pull/33502/commits/471d33091124ec447f7d6f9cbef1c8e918a1b7c3

**Pull request merged**
